### DISCUSSION TITLE
How to detect the network change and update the UI or re-render?

### DISCUSSION THREAD

**Discussion by sanjaybip** [wevm/wagmi#1976] [Q&A]
When I change the network. I want to re render the UI. So if user switch to a wrong network, I want to show wrong network button. And when he connect back to correct network, I want to show the correct UI. I have to manually click on click button or refersh the page.

I used the `const { chain } = useNetwork()` in my code, but not sure how to re render the UI. 
I am using rainbowkit for wallet connection.
![metamaskissue-op](https://user-images.githubusercontent.com/108268573/223993251-7e18631b-f3aa-456f-98d7-69f571024683.gif)

**Comment by sanjaybip** [comment]
So basically on changing the network via metamask, it get disconnected. I need to refersh the page or click on button to see the correct UI (Either wrong network button or connected network with data)

**Comment by OoXooOx** [comment]
useEffect(() => {
        if (chain) {
            if (chain?.id !== goerli.id) switchNetwork?.(goerli.id)
        }
    }, [chain, switchNetwork])

**Reply by sanjaybip** [reply]
Thanks. It doesn't solved the problem exactly. But prompt user to change network once we refresh the page, which is definitely a better UX.
